compiler/shared/main/scala/mlscript/compiler/ClassLifter.scala

```diff
@@ -8,7 +8,8 @@ import scala.collection.mutable.Map as MutMap
 import scala.collection.mutable.Set as MutSet
 import scala.collection.mutable.ArrayBuffer as ArrayBuffer
 import mlscript.codegen.CodeGenError
-import mlscript.compiler.mono.MonomorphError
+
+class MonomorphError(error: String) extends Error(error)
```

Comment:
Why does the class lifter raise a `MonomorphError`? At least rename it to something more generic, such as `CompilerError`.